### PR TITLE
monado: unstable-2024-01-02 -> 24.0.0

### LINKED ISSUE #318177: Update request: monado 2024-01-02 → 2024-06-07
- Package name: monado
- Latest released version: 2024-06-07 (24.0.0)

- Current version on the unstable channel: 2024-01-02
- Current version on the stable/release channel: 2024-01-02

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@Scrumplex 
@expipiplus1
@prusnak

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Scrumplex**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/318177

https://gitlab.freedesktop.org/monado/monado/-/compare/bfa1c16ff9fc759327ca251a5d086b958b1a3b8a...v24.0.0

Looks like ffmpeg was dropped, alongside Basalt and Kimera?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 318203` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>monado</li>
    <li>opencomposite-helper</li>
  </ul>
</details>

**Comment by niklaskorz** [comment]
It looks like Basalt has just been replaced with a more generic dynamic library interface, so Basalt now has to be compiled separately and loaded through an environment variable:

https://gitlab.freedesktop.org/monado/monado/-/blob/main/src/xrt/auxiliary/tracking/t_tracker_slam.cpp#L48

https://gitlab.freedesktop.org/mateosss/basalt

> If you want to run OpenXR application with Monado, you need to set the
environment variable VIT_SYSTEM_LIBRARY_PATH to the path of the basalt library.
By default, Monado will try to load the library from /usr/lib/libbasalt.so if
the environment variable is not set.

**Comment by Scrumplex** [comment]
I am sure #295107 would just need to add its library to LD_LIBRARY_PATH (or perhaps even RPATH) to make it work then :D

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5840,7 +5840,7 @@ with pkgs;
 
   monetdb = callPackage ../servers/sql/monetdb { };
 
-  monado = callPackage ../applications/graphics/monado {
+  monado = callPackage ../by-name/mo/monado/package.nix {
     inherit (gst_all_1) gstreamer gst-plugins-base;
   };
```
Comment (Pandapip1) [inline pkgs/top-level/all-packages.nix:5845]:
```suggestion
```

Packages in `by-name` are automatically called.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Nit: Now that we're using versioned releases, we should add `nix-update-script`

https://ryantm.github.io/nixpkgs/stdenv/stdenv/#:~:text=A%20common%20pattern%20is%20to%20use%20the%20nix%2Dupdate%2Dscript%20attribute%20provided%20in%20Nixpkgs%2C%20which%20runs%20nix%2Dupdate%3A

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (Scrumplex) [reply pkgs/top-level/all-packages.nix:5845]: This is here as the package function expects `gstreamer` `gst-plugins-base` as package arguments.

Comment (Pandapip1) [reply pkgs/top-level/all-packages.nix:5845]: You can use `gst_all_1` as the argument name.

**Review by Pandapip1 — APPROVED** [review]
LGTM

**Comment by loco-choco** [comment]
> I am sure #295107 would just need to add its library to LD_LIBRARY_PATH (or perhaps even RPATH) to make it work then :D

Oh! Nice to hear that! I updated the package to use the latest release of basalt, so I guess we would just need to make a simple module to add it to `LD_LIBRARY_PATH` like in https://github.com/NixOS/nixpkgs/blob/cd7b95ee3725af7113bacbce91dd6549cee58ca5/nixos/modules/system/boot/timesyncd.nix#L58, right?

**Pull request merged**